### PR TITLE
[TS] LPS-156003 Uploaded document is lost after changing Document Type in Documents and Media

### PR DESCRIPTION
**Pull request by orsolyaDekany**
Hi Team,

LPS: https://issues.liferay.com/browse/LPS-156003
PTR: https://issues.liferay.com/browse/PTR-3199

image

Could you please review it?

cc. @robertoDiaz 

Thank you.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp [hunk 1]
```diff
@@ -669,6 +677,7 @@ renderResponse.setTitle(headerTitle);
 	function <portlet:namespace />changeFileEntryType() {
 		Liferay.Util.setFormValues(form, {
 			<%= Constants.CMD %>: '<%= Constants.PREVIEW %>',
+			documentUpload: true,
```
[the remark this answers is not in the document]
Comment (orsolyaDekany) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp:680]: Yes, the intention is that the message appears every time the Doc Type is changed.

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 2]
```diff
@@ -5352,6 +5352,7 @@ document-extensions-do-not-match=Document extensions do not match.
 document-fields=Document Fields
 document-library-field-type-description=Send documents and media files.
 document-library-field-type-label=Upload
+document-may-have-been-lost=Document may have been lost. Please upload it again.
```
[the remark this answers is not in the document]
Comment (orsolyaDekany) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:5355]: Ok. Maybe this one is better: "The uploaded document may have been cleared. Please upload it again."

[the remark this answers is not in the document]
Comment (toomanycurls) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:5355]: If the file is lost any time they change the doc type, let's say "The uploaded document was cleared. Please upload it again."

[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:5355]: One thing to make it clear:

On this case:

> 
> Reproduction steps:
> 
> 1. Start Master or 7.3.x and login
> 2. Create a Document Type
> 3. Create a folder in Documents and Media
> 4. Go into the folder and choose File Upload
> 5. Choose a file to upload
> 6. Publish
> 7. go to the folder
> 8. Edit document
> 9. Change Document Type (the page will be reloaded)

on this case the file does not is deleted on my tests. or I'm wrong?

So in this case we are telling the user that the file may be eliminated, when is not. So this fix only is useful when the file is new (fileEntry == null for example) or we change the uploaded file (re-upload a new one) and then change the document type.

For us, this message is confuse and imprecise in the way this fix is made. We can talk about it if you want

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp [hunk 3]
```diff
@@ -261,6 +261,14 @@ renderResponse.setTitle(headerTitle);
 						</div>
 					</c:if>
 
+					<aui:input name="documentUpload" type="hidden" value="<%= false %>" />
```
[the remark this answers is not in the document]
Comment (boton) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_file_entry.jsp:264]: Yep, `documentUploaded` sounds better to me.

**Review by boton — CHANGES_REQUESTED** [review]

**Comment by AliciaGarciaGarcia** [comment]
closed in favor of https://github.com/liferay-lima/liferay-portal/pull/3315

**Pull request closed without merging**